onivim/oni2
# feat(menubar/#1255): Initial menu implementation for Windows/Linux

WIP for an initial menubar implementation ( related #1255 ) - will be using https://github.com/revery-ui/revery/pull/1024 for OSX, and rendering them via Revery for Linux / Windows.

__TODO:__
- [x] Add configuration options (configuration, choose theme, keybindings) - preferences submenu
- [x] Style and improve shortcut key rendering (separate element as part of the `item` in the context menu)
- [x] Add arrow codicon for submenu
- [x] Test styling on Windows / Linux
- [x] Fix windows padding
- [x] Fix mouse-over behavior when already open
- [x] Add help items: `Welcome`, `Version`, `Documentation`, `DevTalk`, `Twitter`/`Discord`, `Log Issue`
- [x] Remove red debug squares / background
- [x] Add edit items: Find-in-files
- [x] Add view items: Sidebar (move to group, fix titles)
- [x] Add view items: Command Palette
- [x] Add view: Zoom -> Zoom In / Zoom out/ Zoom reset

__Next steps:__
- Iteratively add more menu items via contributions
- Wire up the OSX implementation
- Add view items: Window -> Split Left/Right/Down/Up Move Left/Right/Up/Down
- Add view items: Pane (Messages, Quickfix, Output)
- [x] Check keybindings for shortcut keys
